index.js

```diff
@@ -90,8 +93,49 @@ function handleClose (writeStream) {
       writeStream.emit('close')
     })
   }
+  function trapWindowsEPERMRename (err) {
+    if (err.syscall && err.syscall === 'rename' &&
+        err.code && err.code === 'EPERM' &&
+        writeStream.__isWin
+    ) {
+      if (fs.existsSync(writeStream.__atomicTarget)) {
+        var tmpHash = crypto.createHash('sha256')
+        var targetHash = crypto.createHash('sha256')
+
+        fs.readFile(writeStream.__atomicTmp,
```

Comment:
I would also prefer to not have a cascade of callbacks here and instead use named functions for each layer of callbacks. Keeps the code flatter and the callback names have a self-documenting property.